experimental/dds/tree2/src/test/feature-libraries/optional-field/optionalField.spec.ts

```diff
@@ -362,24 +382,27 @@ describe("optionalField", () => {
 				};
 
 				const expected: OptionalChangeset = {
-					fieldChange: {
-						id: brand(1),
-						wasEmpty: true,
-						newContent: {
-							set: { type: brand("value"), value: "X" },
-							buildId: { localId: brand(41) },
-						},
-					},
-					childChanges: [[{ localId: brand(0) }, arbitraryChildChange]],
+					build: [
+						{ id: { localId: brand(41) }, set: { type: brand("value"), value: "X" } },
+					],
+					// TODO:AB#6298: This test case demonstrates a problem with rebasing transactions:
+					// we don't realize that { localId: brand(1) } no longer refers to the right node
+					// because we rebased over a change that detaches that node. We either need to augment
+					// this with a move from { localId: brand(0), revision: taggedBaseChange.revision } => { localId: brand(1) }
+					// OR update the child change here to refer to { localId: brand(0), revision: taggedBaseChange.revision }!
+					// Right now we do things inconsistently with 'self' due to how renamedDsts works in optional field, which causes this bug.
+					moves: [[{ localId: brand(41) }, "self", "nodeTargeting"]],
+					childChanges: [[{ localId: brand(1) }, arbitraryChildChange]],
+					reservedDetachId: { localId: brand(1) },
```

Comment:
Maybe mention that `reservedDetachId` shound not be set here.

Reply:
I think it depends on which strategy we use to resolve the above TODO. I'm leaning toward `expected`'s child changes referring to `{ localId: 0 }` and keeping reservedDetachId, but haven't thought super deep about it since writing this comment. I'll leave a note though. That should make it clear enough the expectation here isn't desired long-term :)